src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,103 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int SIZE = 10;
+    private T[] arrayList;
+    private int count;
+
+    ArrayList() {
+        arrayList = (T[]) new Object[SIZE];
+        count = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        growArray();
+        arrayList[count] = value;
+        count++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        growArray();
+        if (index < count) {
```

Comment:
```suggestion
        if (index < count && index >=0) {
```
скрызь такы перевырки бажано додати)